src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -496,12 +496,21 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                                             win32ResourceStreamOpt,
                                             diagnosticBag,
                                             cancellationToken);
+                                    }
 
-                                        if (success)
-                                        {
-                                            compilation.ReportUnusedImports(null, diagnosticBag, cancellationToken);
-                                        }
+                                    var usingDiagnostics = DiagnosticBag.GetInstance();
+
+                                    // call ReportUnusedImports unconditionally
+                                    // since that also signals the event queue about completion of compiling.
+                                    compilation.ReportUnusedImportsAndFinishCompiling(null, usingDiagnostics, cancellationToken);
```

Comment:
Could the calls to `CompleteTree` be split out to a separate method from `ReportUnusedImports` so `ReportUnusedImports` is still only called in the `success` case but `CompleteTree` is called always?